src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,284 @@
+            return compilation.ClassifyCommonConversion(type, disposableType).IsImplicit;
+        }
+
+        private async Task<Document> IntroduceUsingStatementAsync(
+            Document document,
+            TextSpan span,
+            CancellationToken cancellationToken)
+        {
+            var (declarationStatement, localVariable) = await FindDisposableLocalDeclaration(document, span, cancellationToken).ConfigureAwait(false);
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
+
+            var statementsToSurround = GetStatementsToSurround(declarationStatement, localVariable, semanticModel, syntaxFactsService, cancellationToken);
+
+            // Separate the newline from the trivia that is going on the using declaration line.
+            var trailingTrivia = SplitTrailingTrivia(declarationStatement, syntaxFactsService);
+
+            var usingStatement =
+                CreateUsingStatement(
+                    declarationStatement,
+                    trailingTrivia.sameLine,
+                    statementsToSurround)
+                    .WithLeadingTrivia(declarationStatement.GetLeadingTrivia())
+                    .WithTrailingTrivia(trailingTrivia.endOfLine);
+
+            if (statementsToSurround.Any())
+            {
+                var parentStatements = GetStatements(declarationStatement.Parent);
+                var declarationStatementIndex = parentStatements.IndexOf(declarationStatement);
+
+                var newParent = WithStatements(
+                    declarationStatement.Parent,
+                    new SyntaxList<TStatementSyntax>(
+                        parentStatements.Take(declarationStatementIndex)
+                        .Concat(new[] { usingStatement })
+                        .Concat(parentStatements.Skip(declarationStatementIndex + 1 + statementsToSurround.Count))));
+
+                return document.WithSyntaxRoot(root.ReplaceNode(declarationStatement.Parent, newParent
+                    .WithAdditionalAnnotations(Formatter.Annotation)));
+            }
+            else
+            {
+                return document.WithSyntaxRoot(root.ReplaceNode(declarationStatement, usingStatement
+                    .WithAdditionalAnnotations(Formatter.Annotation)));
+            }
+        }
+
+        private SyntaxList<TStatementSyntax> GetStatementsToSurround(
+            TLocalDeclarationSyntax declarationStatement,
+            ILocalSymbol localVariable,
+            SemanticModel semanticModel,
+            ISyntaxFactsService syntaxFactsService,
+            CancellationToken cancellationToken)
+        {
+            if (CanRefactorToContainBlockStatements(declarationStatement.Parent))
```

Comment:
it seems like this check would always return true... no?  wasn't this checked already in the analysis phase?